extensions/wikia/Discussions/maintenance/dumpForumData.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Dumps the Forum data for the site selected by setting the SERVER_ID variable
+ *
+ * @ingroup Maintenance
+ */
+
+error_reporting(E_ALL);
+
+require_once( __DIR__ . '/../../../../maintenance/Maintenance.php' );
+include_once( __DIR__ . '/ForumDumper.php' );
+
+class DumpForumData extends Maintenance {
+	/** @var  \Discussions\ForumDumper */
+	private $dumper;
+	private $fh;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Dumps a set of INSERT statements suitable for importing into Discussion "import" tables';
+		$this->addArg( 'out', "Output file for SQL statements", $required = false );
+	}
+
+	public function execute() {
+		if ( $this->hasOption( 'out' ) ) {
+			$this->fh = fopen( $this->getArg(), 'w' );
+		} else {
+			$this->fh = STDOUT;
+		}
+		$this->dumper = new Discussions\ForumDumper();
+
+		$this->clearImportTables();
+		$this->dumpPages();
+		$this->dumpRevisions();
+		$this->dumpVotes();
+	}
+
+	public function clearImportTables() {
+		fwrite( $this->fh, "DELETE FROM import_page;\n" );
+		fwrite( $this->fh, "DELETE FROM import_revision;\n" );
+		fwrite( $this->fh, "DELETE FROM import_vote;\n" );
+	}
+
+	public function dumpPages() {
+		$pages = $this->dumper->getPages();
+
+		foreach ( $pages as $id => $data ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$id" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)